### PR TITLE
Ensure that ported analyzers for FxCop Naming rules analyze only externally visible symbols

### PR DESCRIPTION
**Pull request by mavasani**
Addresses https://github.com/dotnet/roslyn-analyzers/issues/1432#issuecomment-349579767

### CODE REVIEW

**Comment by mavasani** [comment]
Ping?

File: src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumsShouldHavePluralNames.cs [hunk 1]
```diff
@@ -108,7 +108,7 @@ public override void Initialize(AnalysisContext analysisContext)
         private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol flagsAttribute)
         {
             var symbol = (INamedTypeSymbol)context.Symbol;
-            if (symbol.TypeKind != TypeKind.Enum)
+            if (symbol.TypeKind != TypeKind.Enum || !symbol.IsExternallyVisible())
```
Comment (heejaechang) [inline src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumsShouldHavePluralNames.cs]: what happens for protected? is this right definition for IsExternallyVisible?

http://index/?query=IsExternallyVisible&rightProject=Analyzer.Utilities&file=Extensions%5CISymbolExtensions.cs&rightSymbol=rccoswinkudx

Comment (mavasani) [reply src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumsShouldHavePluralNames.cs]: This is the definition for this method: https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Extensions/ISymbolExtensions.cs#L91

Comment (heejaechang) [reply src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumsShouldHavePluralNames.cs]: I see, so it treat protected as public.

**Pull request merged**